### PR TITLE
Added frequency list

### PR DESCRIPTION
**Pull request by Vilhelm-Ian**
For a brief moment let me brag that this is my first time working with python and Qt

### CODE REVIEW

**Comment by Vilhelm-Ian** [comment]
A small problem does exist. If the file specified by the user contains a space it won't work. I didn't want to bother with that since they can and should rename the file with _

**Comment by mortii** [comment]
Very nice! I'll go through it tomorrow.

File: ankimorphs/frequency_file_generator.py
Comment (mortii) [inline ankimorphs/frequency_file_generator.py:1]: Someone created an issue in the morman repo a while ago where they said the readability analyzer freezes. This was not apparent if you only used a couple of small input files, but if you have hundreds or thousands or files (which would be a normal use-case) then anki will freeze, potentially for a very long time. 

To fix this we need to run the generating part in a background thread by using QueryOp like we do in recalc:
https://github.com/mortii/anki-morphs/blob/5f4336bc883c889048caeb49261afc00734b1866/ankimorphs/recalc.py#L35-L48

Here are the Anki API Background Operations docs: https://addon-docs.ankiweb.net/background-ops.html

File: ankimorphs/frequency_file_generator.py [hunk 1]
```diff
@@ -43,20 +49,64 @@ def _setup_buttons(self) -> None:
...
+                file_path = self.path.joinpath(file)
+                if not Path(file_path).is_file():
+                    tooltip(str(file_path) + " dosen't exist", parent=mw)
+                    return None
+                with open(file_path, mode="r", encoding="utf-8") as file:
+                    return file.read()
 
-        tooltip("clicked create frequency file button", parent=mw)
+    def _generate_frequency_list(self, morphes: list[Morpheme]) -> list[[str, str, int]]:
+        hash_map = {}
```
Comment (mortii) [inline ankimorphs/frequency_file_generator.py:101]: a map means something else in python compared to most other programming languages (a map object is an iterator). This should be renamed to 'morph_occurrence_dict' or something to that affect.

**Review by mortii — CHANGES_REQUESTED** [review]
A couple of tweak suggestions

File: ankimorphs/frequency_file_generator.py (hunk 1 above)
Comment (Vilhelm-Ian) [reply ankimorphs/frequency_file_generator.py:101]: okay will get around to them

Comment (Vilhelm-Ian) [reply ankimorphs/frequency_file_generator.py]: @mortii I added the changes

Comment (mortii) [reply ankimorphs/frequency_file_generator.py]: @Vilhelm-Ian ok, i'll test it.

**Comment by mortii** [comment]
I think we should probably support files that have whitespaces in the name.

If someone were to download 1000+ one piece srt files, all of which has 6 whitespaces in the name, then changing them would be a non-trivial task for the user. Even if they know how to do it with a script it would still be inconvenient.

When trying to test this PR I just downloaded a random srt file, and when copying it a bunch of times they get names like this: 'jujutsu_kaisen (2nd Copy).srt', which causes an error. 

Having whitespaces in the file name might be the norm rather than the exception? I dunno, but even if it is less common I think we should probably support it.

Thoughts @Vilhelm-Ian  ?

**Comment by Vilhelm-Ian** [comment]
You are right. I am going to try and fix it in the following days

**Comment by Vilhelm-Ian** [comment]
@mortii  I fixed it was a lot simpler than I thought :)

**Comment by mortii** [comment]
@Vilhelm-Ian nice. I'll merge it into a dev branch and fix the pre-commit errors and trim it up a little bit :+1:

**Pull request merged**